app/components/ChartEditor/index.js

```diff
@@ -10,6 +10,28 @@ import { appSteps } from '../../constants/appSteps';
 import * as styles from './ChartEditor.css';
 
 export default class ChartEditor extends AppComponent {
+  constructor() {
+    super();
+    this._updateDimensions = this._updateDimensions.bind(this);
+  }
+
+  componentDidMount() {
+    window.addEventListener('resize', this._updateDimensions);
+    this._updateDimensions();
+  }
+
+  _updateDimensions() {
+    const subCompWidth =
+      document.querySelector('[class*=subcomponentContainer]').clientWidth;
+    const offsetLeft = document.querySelector('[class*=appComponent]').offsetLeft;
```

Comment:
You could assign the result of selecting class appComponent as a variable, as a micro optimization, since you are doing it twice, but it would also help with your line length here.